### PR TITLE
Experiment with integrals by indexed symbols.

### PR DESCRIPTION
**Pull request by Upabjojr**

### CODE REVIEW

**Comment by Upabjojr** [comment]
https://github.com/sympy/sympy/issues/9695

**Comment by Upabjojr** [comment]
@bjodah @moorepants
I have not yet introduced the contraction of identical symbols with different indices.

You may have a look, the code is quite compact.

File: sympy/tensor/indexed_integrals.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+from sympy.tensor import Idx, IndexedBase, Indexed
+from sympy import Integral, Dummy, sympify, Tuple
+
+
+class _IndexedIntegral(Integral):
+    """
+    Experimental class to test integration by indexed variables.
```
Comment (bjodah) [inline sympy/tensor/indexed_integrals.py]: Even though it is experimental maybe we want some Examples in the docstring? I always found SymPy's larg number of examples to be very useful.

**Comment by bjodah** [comment]
Sorry for the down time, I've been busy. I think it looks good, since it is declared private I guess we could merge this as is.
The removal of the specialization of args in IndexBase also looks sensible to me.

**Comment by Upabjojr** [comment]
I have still to write the partitioning algorithm part. Or can we do it in another PR?

**Comment by bjodah** [comment]
If your current function gets a proper docstring (also describing that it currently assumes all `Idx` instances to differ from each other), I think this is good to merge, although I don't know about the SOPs for including this kind of "provisional" features. So someone more senior in the project would need to give their view here.. maybe an email on the mailing list?

**Comment by Upabjojr** [comment]
Being private, one could simply delete this feature, as no end-users are supposed to be informed about this code (docstring is not linked to SymPy's documentation builder).

**Comment by Upabjojr** [comment]
> I think this is good to merge, although I don't know about the SOPs for including this kind of "provisional" features. So someone more senior in the project would need to give their view here.. maybe an email on the mailing list?

@certik what do you think? This PR introduces an experimental private class, it is unlinked from the documentation, so end-users are unlikely to stumble upon this by chance. Can we merge?

**Comment by moorepants** [comment]
I'm fine to merge if it in experimental. Some projects have a "sandbox" package for experimental stuff that is explicitly designated as having an unreliable API. We could set one up here for that?

**Comment by Upabjojr** [comment]
@moorepants what do you mean exactly? Do you want me to create a _sympy.sandbox_ module and put experimental stuff there?

Actually that sounds a good idea.

**Comment by moorepants** [comment]
Yes, that is what I mean. I think that the group would generally be fine with that. We would just need to have clear documentation that explains that.

**Comment by leosartaj** [comment]
+1

**Comment by Upabjojr** [comment]
@moorepants here you have _sympy.sandbox_, I put a short description in _\_\_init\_\_.py_.

Tell me if you like it this way.

**Comment by bjodah** [comment]
Look good to me, in this case, do we still need to prepend the class name with an underscore? (One could argue that we want the classes and functions in the sandbox module to be included in the documentation)

**Comment by moorepants** [comment]
I agree with @bjodah that the leading underscore is now unnecessary since we are going to introduce the sandbox module. +1 for merging if that is fixed.

**Comment by Upabjojr** [comment]
Here it is.

I think the failure is spurious, I didn't touch on _Relational_.

**Comment by bjodah** [comment]
Great! merging, and merry Christmas!

**Comment by Upabjojr** [comment]
Merry Christmas!

**Pull request merged**